src/main/java/book/store/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+import book.store.onlinebookstore.repository.order.item.OrderItemRepository;
+import book.store.onlinebookstore.repository.shoppingcart.ShoppingCartRepository;
+import book.store.onlinebookstore.repository.user.UserRepository;
+import book.store.onlinebookstore.service.OrderService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserRepository userRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderDto placeOrder(Long userId, CreateOrderRequestDto requestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("Your shopping cart is empty"));
+        Set<CartItem> cartItems = shoppingCart.getCartItems();
+
+        Order newOrder = initNewOrder(requestDto, userId);
+        orderRepository.save(newOrder);
+        BigDecimal total = BigDecimal.ZERO;
+        Set<OrderItem> orderItems = cartItems.stream()
+                .map(cartItem -> getOrderItem(cartItem, newOrder, total))
+                .collect(Collectors.toSet());
+        newOrder.setOrderItems(orderItems);
+
+        shoppingCartRepository.deleteById(userId);
+        return orderMapper.toDto(newOrder);
+    }
+
+    @Override
+    public List<OrderDto> getOrders(Long userId, Pageable pageable) {
+        List<Order> allByUserId = orderRepository.findAllByUserId(userId, pageable);
+        return allByUserId.stream().map(orderMapper::toDto).toList();
+    }
+
+    @Override
+    @Transactional
+    public List<OrderItemDto> getOrderItemsByOrderId(Long userId, Long orderId) {
+        Order order = orderRepository.findByIdAndUserId(orderId, userId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find order by id "
+                        + orderId));
+        return order.getOrderItems().stream().map(orderItemMapper::toDto).toList();
+    }
+
+    @Override
+    public OrderItemDto getItemById(Long userId, Long orderId, Long itemId) {
+        Order order = orderRepository.findByIdAndUserId(orderId, userId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find order by id "
+                        + orderId));
+        OrderItem orderItem = orderItemRepository.findByIdAndOrderId(itemId, order.getId())

```

Comment:
you already have orderId in your method parameters, so don't need to find it in DB

Reply:
I am doing it because if user type someone`s else order id, he will receive id, so i firstly check if this is that user order and then responsing OrderItem